features/FEATURE_BLE/tests/generic/GattClient/TestDiscoverAllServices.cpp

```diff
@@ -0,0 +1,722 @@
+				AttErrorResponse(opcode, error_code)
+			);
+			return BLE_ERROR_NONE;
+		});
+	}
+
+	/*
+	 * Set service discovery expectation when the procedure discover all services
+	 * is used.
+	 */
+	void set_discover_all_services_expectations() {
+		auto services_transactions = get_services_transactions();
+
+		uint16_t next_attribute_handle = 0x0001;
+		for (const auto& service_transactions : services_transactions) {
+			//Log::info() << "expect discover_primary_service(" << connection_handle << ", " << next_attribute_handle << ")" <<std::endl;
+			// set expectation on the discovery function
+			EXPECT_CALL(
+				mock_client, discover_primary_service(connection_handle, next_attribute_handle)
+			).WillOnce(
+				reply_read_by_group_type(service_transactions)
+			);
+			if (!has_characteristic_cb) {
+				set_services_callback_expectations(service_transactions);
+			}
+
+			next_attribute_handle = service_transactions.back().end + 1;
+		}
+
+		// if counter didn't wrap then send an error
+		if (next_attribute_handle != 0) {
+			// set expectation on the discovery function
+			//Log::info() << "expect discover_primary_service(" << connection_handle << ", " << next_attribute_handle << ") "
+			//<< "to fail with opcode = " <<  AttributeOpcode::READ_BY_GROUP_TYPE_REQUEST << " and error = " << AttErrorResponse::ATTRIBUTE_NOT_FOUND << std::endl;
+			EXPECT_CALL(
+				mock_client, discover_primary_service(connection_handle, next_attribute_handle)
+			).WillOnce(
+				reply_error(AttributeOpcode::READ_BY_GROUP_TYPE_REQUEST, AttErrorResponse::ATTRIBUTE_NOT_FOUND)
+			);
+		}
+	}
+
+	/*
+	 * Set client expectations when the discover services by UUID is used.
+	 */
+	void set_discover_services_by_uuid_expectations() {
+		std::vector<std::vector<uint8_t>> services_response;
+		auto services_transactions = get_services_transactions();
+
+		uint16_t next_attribute_handle = 0x0001;
+		for (const auto& transaction : services_transactions) {
+
+			//Log::info() << "expect discover_primary_service_by_service_uuid(" << connection_handle << ", " << next_attribute_handle << "," << service_filter << ")" <<std::endl;
+			// set expectation on the discovery function
+			EXPECT_CALL(
+				mock_client, discover_primary_service_by_service_uuid(
+					connection_handle,
+					next_attribute_handle,
+					service_filter
+				)
+			).WillOnce(
+				reply_find_by_type_value(transaction)
+			);
+
+			if (!has_characteristic_cb) {
+				set_services_callback_expectations(transaction);
+			}
+			next_attribute_handle = transaction.back().end + 1;
+		}
+
+		// note if last service end handle was 0xFFFF do nothing
+		// otherwise set a discovery expectation and reply with an error
+		// if counter didn't wrap then send an error
+		if (next_attribute_handle != 0) {
+			// set expectation on the discovery function
+			//Log::info() << "expect discover_primary_service_by_service_uuid(" << connection_handle << ", " << next_attribute_handle << "," << service_filter << ") "
```

Comment:
Why these log lines are commented out?

Reply:
They can be removed; they are a residue of test developments.

Reply:
Assuming that the log comments won't be removed.

If they still will be, then we can restart CI once that commit is in.